docs/02-guides/display-wysiwyg-content.mdx

```diff
@@ -0,0 +1,344 @@
+        "Magento2/Catalog/Product" // ensure that you can fetch a product in your Wysiwyg data
+      ],
+      // highlight-end
+    ```
+1.  Define a new `WysiwygProductNameData` type that implements `WysiwygNodeData`
+    and expose all the data you need to display your component
+    ```ts title="extensions/acme-extension/modules/wysiwyg/index.ts"
+    typeDefs: /* GraphQL */ `
+      type WysiwygProductNameData implements WysiwygNodeData {
+        dataId: ID
+        product: Product
+      }
+    `,
+    ```
+1.  Setup the resolvers to tell GraphQL how to fetch the `product` field in your
+    new `WysiwygProductNameData` type
+    ```ts title="extensions/acme-extension/modules/wysiwyg/resolvers.js"
+    export default {
+      WysiwygProductNameData: {
+        product: ({ node }, _, { loaders }) => {
+          const productSkuAttribute = node.attrs.find(
+            ({ name }) => name === "sku"
+          );
+          if (!productSkuAttribute) {
+            return null;
+          }
+          return loaders.Product.load(productSkuAttribute.value);
+        },
+      },
+    };
+    ```
+1.  Register the HTML tag and associate it with the GraphQL type
+    ```ts title="extensions/acme-extension/modules/wysiwyg/runtime.ts"
+    // ...
+    contextEnhancer: ({ loaders }) => {
+      loaders.Wysiwyg.registerNodeType(
+        "product-name", // the tag name in your HTML
+        "WysiwygProductNameData" // The associated GraphQL type name
+      );
+    };
+    // ...
+    ```
+
+#### Fetch the data to make it available in your custom Product Name component
+
+1. Override the `WysiwygFragment.gql` to fetch the new `WysiwygProductNameData`
+
+   ```graphql title="theme/modules/Wysiwyg/WysiwygFragment.gql"
+   #import "theme/modules/Wysiwyg/MagentoWysiwyg/MagentoWysiwygFragment.gql"
+
+   fragment WysiwygFragment on Wysiwyg {
+     childNodes
+     data {
+       dataId
+       // highlight-start
+       ... on WysiwygProductNameData {
+         product {
+           sku
+           name
+         }
+       }
+       // highlight-end
+     }
+     ... on MagentoWysiwyg {
+       ...MagentoWysiwygFragment
+     }
+   }
+   ```
+
+   Feel free to split this in a Fragment dedicated to your `CustomWysiwyg` just
+   like it is done in the core for `MagentoWysiwyg`.
+
+2. Reference the `product-name` tag in your componentMap and use the new `data`
+   prop available in the mapped component.
+
+   ```ts title="extensions/acme-extension/theme/modules/Wysiwyg/DefaultWysiwyg.js"
+   const defaultComponentsMap = {
+     product-name: (props) => <span>{props.data.product.name}</span>
+   }
+   ```
+
+   :::note Important
+
+   Please keep in mind that for simplicity's sake, the product name component is
+   not code split, but if it grows in complexity, please make sure to code split
+   it using <code>@loadable/component</code> to avoid too large bundles.
+
+   :::
```

Comment:
We no longer support loadable, link to an example using react `lazy()` and `Suspense`